### PR TITLE
Nodemailer Email Styling

### LINKED ISSUE #15: Style nodemailer emails
The following emails need to be formatted:
- createReservation: reservation confirmation emails are sent to guest and host
- createListing: listing confirmation email is sent to host
- signUp: account verification email is sent to host before they can sign up
- activate/deactivate: check in/out emails sent to guest and host

### PR DESCRIPTION
**Pull request by nishantbalaji**
# Relevant issue

Closes #15 

# Summary of change

* I accidentally branched from `friend-requests` so please refer to that too (#142 )
* Also let me know if I missed any emails
* Updated formatting for following emails: 
   * Listing Expiration
   * Account Verification
   * Listing Creation Confirmation
   * Transfer Invite
   * Transfer Acceptance (2 emails: one to sending and one to receiving party)
   * Transfer Rejection (2 emails: one to sending and one to receiving party)
   * Reservation Confirmation for Guest
   * Reservation Confirmation for Host
   * Checkin for Guest
   * Checkin for Host
   * Checkout for Guest
   * Checkout for Host
* Everything is a function for easy access. Pass `HTMLOptions` to `getHTML` to get the formatted HTML, and call `getAttachments` to get the attachments for the email. 
* Updated husky precommit hook for backend to not call `git add .` anymore
* Rebased with master

# Testing/Verification

* navigate to the `emailtester.js` file in `/helpers` directory. 
* change the name and email to your own name and email
* run `node emailtester.js` and check your inbox 
* emails should be the same format as the template, in `template.html`. 
     * links should be white on the gray background with no highlighting
     * you should get 12 emails total 
* please also test on mobile and other email clients
* check that i didnt break anything else please

### CODE REVIEW

**Review by aidenszeto — CHANGES_REQUESTED** [review]
Looks beautiful, just need to update the old nodemailer emails to use the new template

image

**Review by syd-tan — APPROVED** [review]

**Comment by aidenszeto** [comment]
`friend-requests` branch is now merged to master and deleted. I think we can just rebase from master and merge this when it's ready?

**Comment by nishantbalaji** [comment]
> `friend-requests` branch is now merged to master and deleted. I think we can just rebase from master and merge this when it's ready?
@aidenszeto 

It's been done

**Review by syd-tan — APPROVED** [review]
Got all 12 emails, looks good to me!

File: helpers/emailtester.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  sendReservationConfirmationHost,
+  sendCheckinGuest,
+  sendCheckinHost,
+  sendCheckoutGuest,
+  sendCheckoutHost,
+} = require("./emails.helper.js");
+
+// Replace this with your own email address and name
+
+const email = "nishantbalaji30@gmail.com";
```
Comment (aidenszeto) [inline helpers/emailtester.js:18]: Can we set this to contact@visitnomad.com for now?

File: helpers/emails.helper.js [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+    to: email,
+    subject: "You've Been Invited!",
+    html: html,
+    attachments: attachments,
+  };
+
+  sendEmail(userMailOptions);
+};
+
+const sendTransferAccept = (
```
Comment (aidenszeto) [inline helpers/emails.helper.js:118]: I believe we actually send two emails for this, one to the person who sent the transfer and one to the person who accepted the transfer

File: helpers/emails.helper.js [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+    to: mailTo,
+    subject: `Your Transfer was Successful!`,
+    html: html,
+    attachments: attachments,
+  };
+
+  sendEmail(userMailOptions);
+};
+
+const sendTransferRejection = (
```
Comment (aidenszeto) [inline helpers/emails.helper.js:149]: Same here, I think we send two emails to the two parties involved in the transfer

**Review by aidenszeto — COMMENTED** [review]
Getting all 12 emails too, styling looks great. Added a couple comments regarding transfer emails but other than that looks good.

File: helpers/emails.helper.js (hunk 2 above)
Comment (nishantbalaji) [reply helpers/emails.helper.js:118]: You weren't in the first place but I can add it if you want

File: helpers/emails.helper.js (hunk 3 above)
Comment (nishantbalaji) [reply helpers/emails.helper.js:149]: Same as previous comment

**Review by aidenszeto — APPROVED** [review]

**Pull request merged**